Move day-in-the-life to intro (#3858)
quicwg/base-drafts: draft-ietf-quic-transport.md

bidirectional streams allow both endpoints to send data; unidirectional streams
allow a single endpoint to send.  A credit-based scheme is used to limit stream
creation and to bound the amount of data that can be sent.

The QUIC handshake combines negotiation of cryptographic and transport
parameters.  The handshake is structured to permit the exchange of application
data as soon as possible.  This includes an option for clients to send data
immediately (0-RTT), which might require prior communication to enable.

QUIC connections are not strictly bound to a single network path.  Connection
migration uses connection identifiers to allow connections to transfer to a new
network path.

Frames are used in QUIC to communicate between endpoints.  One or more frames
are assembled into packets.  QUIC authenticates all packets and encrypts as much
as is practical.  QUIC packets are carried in UDP datagrams to better facilitate
deployment in existing systems and networks.

{{fig-hs}} shows a simplied handshake and the exchange of packets and frames
that are used to advance the handshake.  Exchange of application data during the

> handshake is enabled where possible, shown with a '*'.

Reviewer: This seems a bit out of place. It would be more useful if it included data exchange and connection termination. Otherwise, I would suggest removing it entirely (this para and the text below as well).

* Authenticated and encrypted header and payload

QUIC establishes a connection, which is a stateful interaction between a client
and server. The primary purpose of a connection is to support the exchange of
data by an application protocol.

> Streams ({{streams}}) are the primary means by which an application protocol

Reviewer 2: Saying 'primary' seems odd given there's no other option in v1.

You could say "Streams are the only means by which an application protocol exchange information in this version of QUIC."

> Streams ({{streams}}) are the primary means by which an application protocol

Reviewer 2: Streams is the only reference in this section, so I'd be inclined to drop it given it's referenced in "Life of a QUIC connection"?

exchanges information.  Streams are ordered sequences of bytes.  Two types of
stream can be created: bidirectional streams allow both endpoints to send data;

> unidirectional streams allow a single endpoint to send.  Endpoints can create
> streams without prior negotiation.  A credit-based scheme is used to limit

Reviewer 2: I'd drop this sentence, since one could consider the stream limit TPs 'negotiation'.

```suggestion
unidirectional streams allow a single endpoint to send.  A credit-based scheme is used to limit
```

stream creation and to bound the amount of data that can be sent.

The QUIC handshake combines negotiation of cryptographic and transport
parameters.  The handshake is structured to permit the exchange of application
data as soon as possible.  This includes an option for clients to send data
immediately (0-RTT), which might require prior communication to enable.

QUIC connections are not strictly bound to a single network path.  Connection
migration uses connection identifiers to allow connections to transfer to a new
network path.

> Frames are used in QUIC to communicate between endpoints.  Multiple frames are
> assembled into packets.  QUIC authenticates all packets and encrypts as much as

Reviewer 2: ```suggestion
Frames are used in QUIC to communicate between endpoints.  One or more frames can
be sent in a single QUIC packet.  QUIC authenticates all packets and encrypts as much as
```

* Authenticated and encrypted header and payload

QUIC establishes a connection, which is a stateful interaction between a client
and server. The primary purpose of a connection is to support the exchange of
data by an application protocol.

Streams ({{streams}}) are the primary means by which an application protocol
exchanges information.  Streams are ordered sequences of bytes.  Two types of
stream can be created: bidirectional streams allow both endpoints to send data;
unidirectional streams allow a single endpoint to send.  Endpoints can create
streams without prior negotiation.  A credit-based scheme is used to limit
stream creation and to bound the amount of data that can be sent.

The QUIC handshake combines negotiation of cryptographic and transport
parameters.  The handshake is structured to permit the exchange of application
data as soon as possible.  This includes an option for clients to send data

> immediately (0-RTT) as possible, which might require prior communication to

Reviewer 3: ```suggestion
immediately (0-RTT), which might require prior communication to
```

enable.

QUIC connections are not strictly bound to a single network path.  Connection
migration uses connection identifiers to allow connections to transfer to a new
network path.

Frames are used in QUIC to communicate between endpoints.  Multiple frames are
assembled into packets.  QUIC authenticates all packets and encrypts as much as

> is practical to avoid incurring a dependency on middleboxes.  QUIC packets are

Reviewer 3: Isn't the dependency the other way around; we're trying to avoid middleboxes becoming dependent on any visible features of QUIC.
